sphinx/themes/basic/static/searchtools.js

```diff
@@ -160,10 +160,15 @@ const Search = {
   _queued_query: null,
   _pulse_status: -1,
 
-  htmlToText: (htmlString) => {
+  htmlToText: (htmlString, anchor) => {
     const htmlElement = new DOMParser().parseFromString(htmlString, 'text/html');
     htmlElement.querySelectorAll(".headerlink").forEach((el) => { el.remove() });
-    const docContent = htmlElement.querySelector('[role="main"]');
+    let docContent = undefined;
+    if (anchor && htmlElement.querySelector(anchor)) {
+      docContent = htmlElement.querySelector(anchor);
+    } else {
+      docContent = htmlElement.querySelector('[role="main"]');
+    }
     if (docContent) return docContent.textContent;
     console.warn(
       "Content block not found. Sphinx search tries to obtain it via '[role=main]'. Could you check your theme or template."
```

Comment:
Could you update this warning message to make sure it's not misleading in cases where an anchor is provided?

Reply:
We only match on the anchor if it's (1) there and (2) querySelector actually matches some content. So I think this warning message is still valid as-is.

Reply:
If we display a result where the hyperlink contains an anchor, and neither the anchor nor the `[role=main]` content are found on the page, then I think it could be confusing to tell a developer/user that `[role=main]` content was missing.

It's partially true, but it doesn't necessarily indicate how to completely resolve the (two-fold) problem.